### DISCUSSION TITLE
Is saving gifs possible?

### DISCUSSION THREAD

**Discussion by KhafraDev** [dlemstra/magick-wasm#40] [Q&A] [answered]
I was wondering if it's currently possible to edit gifs.

```js
import { ImageMagick, initializeImageMagick } from '@imagemagick/magick-wasm';

await initializeImageMagick();

const gif = '...'; // get gif from somewhere

ImageMagick.readCollection(gif, (frames) => {
    // I can loop over each frame, but how do I save?
    for (const frame of frames) {
        frame.charcoal(); // for example
    }

    // ???
    frames.dispose();
});

```

I can read a gif and get an array of frames, but I have no idea how to write/save the frame. I looked at tests and couldn't find much either. Thanks ❤️

**Comment by dlemstra** [comment, marked as answer]
If you want to save the individual frames you could call `frame.write`. And when you want to create a new gif you can call `frames.write`.

**Reply by KhafraDev** [reply]
`frames.write` doesn't seem to be a function, I looked through the typings a little and couldn't find anything there either

```js
TypeError: frames.write is not a function
```

**Reply by dlemstra** [reply]
It seems you are right. Will try to add that this weekend.
